server/controllers/userProfileController.js

```diff
@@ -0,0 +1,66 @@
+const User = require('../models/User')
+const UserProfile = require('../models/UserProfile')
+const asyncHandler = require('express-async-handler')
+
+exports.profile = asyncHandler(async (req, res) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      gender,
+      birthDate,
+      phoneNumber,
+      address,
+      city,
+      country,
+      description,
+      available,
+      availabityPeriod
+    } = req.body
+
+    const profile = await new UserProfile({
+      firstName,
+      lastName,
+      gender,
+      birthDate,
+      phoneNumber,
+      address,
+      city,
+      country,
+      description,
+      available,
+      availabityPeriod
+    })
+    const userProfile = await User.updateOne(
+      { _id: req.params.userId },
+      {
+        $set: {
+          profile
+        }
+      }
+    )
+    return res.status(201).json(userProfile)
+  } catch (err) {
+    return res.status(400).send(err)
+  }
+})
+
+exports.getProfile = asyncHandler(async (req, res) => {
+  try {
+    const userProfile = await User.findById({ _id: req.params.userId }).select(
```

Comment:
Validating the `req.params.userId` prior to passing it as an argument to the `findById` would be ideal :)